Sources/Widgets/Manipulators/LineManipulator/index.js

```diff
@@ -36,17 +36,34 @@ export function projectDisplayToLine(
 // ----------------------------------------------------------------------------
 
 function vtkLineManipulator(publicAPI, model) {
-  // Set our classNae
+  // Set our className
   model.classHierarchy.push('vtkLineManipulator');
 
+  // prettier-ignore
+  publicAPI.getOrigin = () => {
+    if (model.userOrigin) { return model.userOrigin; }
+    if (model.handleOrigin) { return model.handleOrigin; }
+    if (model.widgetOrigin) { return model.widgetOrigin; }
+    return [0, 0, 0]
+  };
+
+  // prettier-ignore
+  publicAPI.getNormal = (camera) => {
```

Comment:
pass in the renderer instead of the camera

Reply:
I've replaced `calldata` by `camera` because we sometimes need to call `manipulator.getNormal` without having access to callData. Example in ShapeWidget/behavior.js:

```js
const computeTextPosition = (worldBounds, textPosition, worldMargin = 0) => {
    const viewPlaneOrigin = model.manipulator.getOrigin();
    const viewPlaneNormal = model.manipulator.getNormal(
      model._renderer.getActiveCamera()
    );
    const viewUp = model._renderer.getActiveCamera().getViewUp();
    // ...
```
How could I do this otherwise?

Reply:
In the ShapeWidget example you shared, it does not seem to be correct to get the "origin"/"normal" from the manipulator.
Text position is not really related to the manipulator but maybe more related ot the handles or the widget itself.

Reply:
I could get it to work using the bounds center as the origin and the camera normal:
```js
const viewPlaneOrigin = [
      (bounds[0] + bounds[1]) * 0.5,
      (bounds[2] + bounds[3]) * 0.5,
      (bounds[4] + bounds[5]) * 0.5,
    ];
const viewPlaneNormal = model._renderer.getActiveCamera().getDirectionOfProjection();
const viewUp = model._renderer.getActiveCamera().getViewUp();
```
I think this makes sense as we always want the text to be centered in the volume (by default) and facing the user camera.

Reply:
+1